RELEASES.md

```diff
@@ -1,5 +1,11 @@
 # Release Notes
 
+## [v1.11.12](https://github.com/ava-labs/avalanchego/releases/tag/v1.11.21)
+
+## Misc
+
+- Added reminder for documenting RELEASE.md in the pull request template
+
```

Comment:
`v1.11.12` has already been released. Also we typically don't include non-user facing things into the release notes.

Reply:
I added this to present an example